### ISSUE TITLE
The `Table.parse` method is not displayed correctly in IDE

### ISSUE THREAD

**Issue by 4e6** [enso-org/enso#6640]
### What type of issue is this?

Transient – Occurring only once

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

Given the expression
```py
operator8.parse 'Sales Timestamp' Value_Type.Date_Time 'dd/MM/yyyy HH:mm:ss'
```

IDE does not display first argument `'Sales Timestamp'`

### Expected behaviour

IDE displays the first argument of `Table.parse` method.

### Screenshots or screencasts

https://github.com/enso-org/enso/assets/357683/b0dde3a2-0dd4-4243-9bf5-40564fd39861

### Enso Version

3dd05c235953954e4f16a6ba9ff638b52d3d98a9

### Browser or standalone distribution

Chromium

### Browser Version or standalone distribution

113.0.5672.63

### Operating System

Linux

**Comment by farmaazon** [comment]
I see the "add first element" button of vector editor. So it seems we expect array there and create ListEditor widget, but the widget badly interprets the string literal (as it have no children, the widget assumes an empty list).

I think we should not allow List Editor widget if the argument has value and it's not a Vector. Or the ListEditor should be able to handle ASTs which are not vector. Or maybe both.

**Comment by Akirathan** [comment]
After looking at the following sources:

From `Table.parse`:
```
@columns Widget_Helpers.make_column_name_vector_selector
```

Widget_Helpers.enso:
```
## PRIVATE
   Make a multiple column name selector.
make_column_name_vector_selector : Table -> Display -> Vector_Editor
make_column_name_vector_selector table display=Display.Always =
    item_editor = make_column_name_selector table display=Display.Always
    Vector_Editor item_editor=item_editor item_default=table.column_names.first.pretty display=display
```

, I believe this is the correct behavior. The only weird behavior that I have observed, is that once the default item (which is the very first column in my case) is selected in the VectorEditor, the dropdown selection for the item cannot be clicked:

[Screencast from 2023-05-11 11-28-52.webm](https://github.com/enso-org/enso/assets/14013887/d57f5bd4-cff9-4c00-83f6-fabd5836e8c6)

cc @jdunkerley is this the expected behavior now, considering the current technical limitations with widgets?

**Comment by Frizi** [comment]
This is a list editor issue that will be fixed with #6540.

On current develop the list editor is incomplete and still under a feature flag. I believe that our feature gate was slightly incorrect, in that it actually allows list editor to show up if the method annotation directly asks for it. Since libs have made progress on it already, I guess this is why it shows up in this unfinished state.
